app/src/main/java/de/tum/in/tumcampusapp/component/tumui/roomfinder/RoomFinderDetailsActivity.kt

```diff
@@ -0,0 +1,300 @@
+
+    lateinit var room: RoomFinderRoom
+    private var mapId: String = ""
+    private var mapsList: List<RoomFinderMap> = ArrayList()
+    private val infoLoaded: Boolean = false
+
+    private var fragment: Fragment? = null
+
+    private var mRoomFinderCoordinateCall: Call<RoomFinderCoordinate>? = null
+    private var mRoomFinderMapsCall: Call<List<RoomFinderMap>>? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        mImageFragment = ImageViewTouchFragment.newInstance()
+        supportFragmentManager.beginTransaction()
+                .add(R.id.fragment_container, mImageFragment)
+                .commit()
+
+        room = intent.getSerializableExtra(EXTRA_ROOM_INFO) as RoomFinderRoom
+
+        startLoading()
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu): Boolean {
+        menuInflater.inflate(R.menu.menu_roomfinder_detail, menu)
+
+        val switchMap = menu.findItem(R.id.action_switch_map)
+        switchMap.isVisible = "10" != mapId && mapsLoaded && fragment == null
+
+        val timetable = menu.findItem(R.id.action_room_timetable)
+        timetable.isVisible = infoLoaded
+        timetable.setIcon(
+                if (fragment == null) R.drawable.ic_outline_event_note_24px
+                else R.drawable.ic_outline_map_24px
+        )
+
+        menu.findItem(R.id.action_directions).isVisible = infoLoaded && fragment == null
+        return true
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        when (item.itemId) {
+            R.id.action_room_timetable -> {
+                getRoomTimetable()
+                invalidateOptionsMenu()
+                return true
+            }
+            R.id.action_directions -> {
+                loadGeo()
+                return true
+            }
+            R.id.action_switch_map -> {
+                showMapSwitch()
+                return true
+            }
+            else -> return super.onOptionsItemSelected(item)
+        }
+    }
+
+    override fun onBackPressed() {
+        // Remove fragment with room timetable if present and show map again
+        if (fragment != null) {
+            getRoomTimetable()
+            invalidateOptionsMenu()
+            return
+        }
+
+        super.onBackPressed()
+    }
+
+    private fun getRoomTimetable() {
+        val ft = supportFragmentManager.beginTransaction()
+        // Remove if fragment is already present
+        if (fragment != null) {
+            ft.replace(R.id.fragment_container, mImageFragment)
+            ft.commit()
+            fragment = null
+            return
+        }
+
+        val roomApiCode = room.room_id
+        fragment = WeekViewFragment.newInstance(roomApiCode)
+        ft.replace(R.id.fragment_container, fragment!!)
+        ft.commit()
+    }
+
+    private fun showMapSwitch() {
+        val list = arrayOfNulls<CharSequence>(mapsList.size)
```

Comment:
This seems very Java-y. Maybe you can replace it with something like this: 

```kotlin
val descriptions = mapsList.map { it.description }
val currentPosition = mapsList.indexOfFirst { it.map_id == mapId }
```

Reply:
That's a lot easier to understand :)